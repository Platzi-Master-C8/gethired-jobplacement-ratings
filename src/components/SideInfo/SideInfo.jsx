import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { Global } from '@emotion/react';
import { styled } from '@mui/material/styles';
import ArrowLeftIcon from '@mui/icons-material/ArrowLeft';
import StarIcon from '@mui/icons-material/Star';
import { Grid, Box, Typography, Button, SwipeableDrawer } from '@mui/material';

import { RatingItem } from '../RatingItem';

import api from '../../services/api';

const globalStyles = (isMobile) => ({
    '.MuiDrawer-root > .MuiPaper-root': {
        height: '30rem',
        top: '20%',
        borderRadius: '20px',
        width: isMobile ? '90%' : '25rem',
    },
});

const GridContainer = styled(Grid)(() => ({
    display: 'flex',
    flexDirection: 'column',
    padding: '30px 30px',
    height: '100%',
    justifyContent: 'space-between',
}));

const GridFlex = styled(Grid)(() => ({
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
}));

const SideInfo = ({ isMobile }) => {
    const [open, setOpen] = useState(false);
    const [isLoaded, setIsLoaded] = useState(false);
    const [review, setReview] = useState({});
    const [error, setError] = useState(null);

    useEffect(() => {
        api.companyEvaluations
            .mockDataOverallReview()
            .then((res) => res)
            .then((data) => {
                setReview(...data);
                setIsLoaded(true);
            })
            .catch((err) => {
                setIsLoaded(true);
                setError(err);
            });
    }, []);

    const toggleDrawer = (newOpen) => () => {
        setOpen(newOpen);
    };

    if (!isLoaded) {
        return <div>Loading...</div>;
    }

    if (error) {
        return <div>Error</div>;
    }

    return (
        <Box>
            <Global styles={globalStyles(isMobile)} />
            <Box sx={{ position: 'fixed', top: '50%', right: '0' }}>
                <Button onClick={toggleDrawer(true)}>
                    <ArrowLeftIcon sx={{ fontSize: '3rem' }} />
                </Button>
            </Box>
            <SwipeableDrawer
                anchor="right"
                open={open}
                onClose={toggleDrawer(false)}
                onOpen={toggleDrawer(true)}
                disableSwipeToOpen={false}
                ModalProps={{
                    keepMounted: true,
                }}
            >
                <GridContainer container>
                    <GridFlex item>
                        <Grid>
                            <Typography variant="h2" fontSize={25} fontWeight={800}>
                                Overall Rating
                            </Typography>
                            <Typography variant="body1" fontSize={16}>
                                Based on {review.total_reviews} reviews
                            </Typography>
                        </Grid>
                        <GridFlex item>
                            <Typography variant="h1" fontSize={25} margin="0 7px">
                                {review.overall_rating}
                            </Typography>
                            <StarIcon sx={{ color: '#edd309', fontSize: '30px' }} />
                        </GridFlex>
                    </GridFlex>
                    <Grid item>
                        <Typography variant="h2" fontSize={17} fontWeight={700}>
                            Rating by category
                        </Typography>
                        <RatingItem title="Job advancements" rating={review.job_advancements} />
                        <RatingItem title="Benefits and perks" rating={review.benefits} />
                        <RatingItem title="Work/life balance" rating={review.work_life_balance} />
                        <RatingItem title="Working Enviorment" rating={review.working_eviorment} />
                        <RatingItem title="Culture" rating={review.culture} />
                    </Grid>
                </GridContainer>
            </SwipeableDrawer>
        </Box>
    );
};

SideInfo.propTypes = {
    isMobile: PropTypes.bool.isRequired,
};

export default SideInfo;
