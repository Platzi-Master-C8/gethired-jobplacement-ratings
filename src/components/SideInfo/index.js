import SideInfo from './SideInfo';
export { SideInfo }