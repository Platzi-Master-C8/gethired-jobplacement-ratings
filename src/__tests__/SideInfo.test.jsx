import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import SideInfo from '../components/SideInfo/SideInfo';

const defaultProps = { isMobile: false };
const renderComponent = (props = {}) => render(<SideInfo {...defaultProps} {...props} />);

describe('component <SideInfo/>', () => {
    test('renders SideInfo', () => {
        const { asFragment } = renderComponent();
        expect(asFragment()).toMatchSnapshot();
    });

    test('renders the button', () => {
        const component = renderComponent();
        component.getByRole('button');
    });

    test('opens the drawer', () => {
        const component = renderComponent();

        fireEvent.click(component.getByRole('button'));

        component.getByText('Overall Rating');
    });
});
