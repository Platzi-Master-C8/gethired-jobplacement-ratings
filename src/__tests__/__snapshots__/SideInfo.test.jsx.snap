// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <SideInfo/> renders SideInfo 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiBox-root css-k9w323"
    >
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1tzp3k3-MuiSvgIcon-root"
          data-testid="ArrowLeftIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m14 7-5 5 5 5V7z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="PrivateSwipeArea-root css-x15wq9"
      style="width: 20px;"
    />
  </div>
</DocumentFragment>
`;
